### ISSUE TITLE
Discussion thread 🔥

### ISSUE THREAD

**Issue by bestsauce** [worstsauce/study#1]
Note: this is a discussion thread only about the way we set things up on Github. Discussions on the book are for their relevant threads, as explained in [README](https://github.com/worstsauce/study/blob/master/README.md).

- Common milestones?

- All exercises or divide them into necessary+optional ones?

- Dividing exercises issues into each section instead of each chapter?

- Working all on a single document for notes?

Testing $\LaTeX$: $\displaystyle\int_0^1 \dfrac{x^2}{x+1}\ \mathrm{d}x$.

$$\boxed{\ \oint_{\partial \Sigma} \mathbf{E} \cdot \mathrm{d}\boldsymbol{\ell}  = - \overset{\color{white}{.}}{\frac{\mathrm{d}}{\mathrm{d}t}} \underset{\color{white}{.}}{\iint_\Sigma} \mathbf{B} \cdot \mathrm{d}\mathbf{S}\ }$$

**Comment by doshgillio** [comment]
Hey there, interested in joining in. What text are you using and is it available online anywhere?

**Comment by BadiGard** [comment]
I'm interested in joining too :)

**Comment by MaxisJaisi** [comment]
Count me in!

**Comment by bestsauce** [comment]
@doshgillio @BadiGard @MaxisJaisi Sent you all an invitation (visit https://github.com/worstsauce to accept the invitation.)

Also is anyone having trouble configuring MathJax with Github?

@doshgillio We'll hopefully work on Sheldon Axler's *Linear Algebra Done Right*

**Comment by doshgillio** [comment]
@bestsauce lovely thank you, I'll source a copy. Would be interesting to know what level people are coming in from I'm a 3rd year maths undergrad who wants to better my understanding of a subject I have previously studied.

**Comment by bestsauce** [comment]
@BadiGard @doshgillio @MaxisJaisi 
Great! It seems now everyone accepted the invitation, now you need to:

1] Open two issues: https://github.com/worstsauce/study/issues/new both titled "Chapter 1". Label one of them with the `Notes` label and with the body:

```
Chapter 1: **Vector Spaces** 

- [ ] $\mathbf{R}^n$ and $\mathbf{C}^n$
- [ ] Definition of Vector Space
- [ ] Subspaces
```


and the other with the `Exercises` label and with content:

```
Chapter 1: **Vector Spaces**

**1.A**

- [ ] Exercise 1
- [ ] Exercise 2
- [ ] Exercise 3
- [ ] Exercise 4
- [ ] Exercise 5
- [ ] Exercise 6
- [ ] Exercise 7
- [ ] Exercise 8
- [ ] Exercise 9
- [ ] Exercise 10
- [ ] Exercise 11
- [ ] Exercise 12
- [ ] Exercise 13
- [ ] Exercise 14
- [ ] Exercise 15
- [ ] Exercise 16

**1.B**

- [ ] Exercise 1
- [ ] Exercise 2
- [ ] Exercise 3
- [ ] Exercise 4
- [ ] Exercise 5
- [ ] Exercise 6

**1.C**

- [ ] Exercise 1
- [ ] Exercise 2
- [ ] Exercise 3
- [ ] Exercise 4
- [ ] Exercise 5
- [ ] Exercise 6
- [ ] Exercise 7
- [ ] Exercise 8
- [ ] Exercise 9
- [ ] Exercise 10
- [ ] Exercise 11
- [ ] Exercise 12
- [ ] Exercise 13
- [ ] Exercise 14
- [ ] Exercise 15
- [ ] Exercise 16
- [ ] Exercise 17
- [ ] Exercise 18
- [ ] Exercise 19
- [ ] Exercise 20
- [ ] Exercise 21
- [ ] Exercise 22
- [ ] Exercise 23
- [ ] Exercise 24
```

(feel free to modify it to remove certain exercises if you want, it's not necessary here to solve all of them)

To set labels:

![](https://raw.githubusercontent.com/worstsauce/study/c0cc1ad25a9d0660e08d4c37e669cc48f2a13fba/screen.png)

After making that we can discuss about the time schedule that you prefer (one chapter per week?) so that we can add milestones that fit here.

@doshgillio I'm a self-studier with no former background on linear algebra, so it's the first time I will study it, what about others? :)

**Comment by MaxisJaisi** [comment]
Dear @bestsauce , I've created my Notes and Exercises issues. Sorry for the delay; there were a few wrinkles on my part to sort out. 

Perhaps we could ask knowledgeable people on Reddit about our MathJax loading problem... ?

I too have virtually no background in LA besides knowing a few things about matrices and linear maps.

**Comment by MaxisJaisi** [comment]
Test: $|a+b| \leq |a| + |b|$
Test II: $x \in \mathbf{R} \implies x^2 \geq 0$

**Comment by bestsauce** [comment]
@MaxisJaisi 

> Perhaps we could ask knowledgeable people on Reddit about our MathJax loading problem... ?

You're having problem with it? First what is your browser?

**Comment by MaxisJaisi** [comment]
@bestsauce Firefox. The same bug you described. MathJax won't load unless I refresh the page.

**Comment by bestsauce** [comment]
@MaxisJaisi Well that's more of a feature that is missing in the script rather than a bug, the only workaround for this now would be to refresh the page or load them in a different tab each time. (I will try to search a bit on this to see if there could be a more useful workaround, perhaps the script would reload itself whenever it hits a different url)

**Comment by MaxisJaisi** [comment]
@bestsauce Also, I'm fairly new to Github. I can see that you've assigned 6 & 7 to me via email, but I can't seem to find the exact notification here; it only says that you've assigned something.

**Comment by MaxisJaisi** [comment]
@bestsauce I see. Noted with thanks.

**Comment by bestsauce** [comment]
@MaxisJaisi Oh, also, you can create your own milestone here: https://github.com/worstsauce/study/milestones

Though it would be better to wait for others so that we can synchronize our work better (with a one chapter per week schedule ideally).

---

Yes, I've assigned these two issues you made because you forgot to assign them to yourself :) https://github.com/worstsauce/study/issues/7 and https://github.com/worstsauce/study/issues/6

**Comment by MaxisJaisi** [comment]
Ah, I see. :) By the way, where do you live, if you don't mind me asking? Our timezone might differ. I'm from Malaysia.

**Comment by bestsauce** [comment]
@MaxisJaisi Our timezones might differ but what's nice about this method is that we don't need to have our clocks synchronized. I'm from the UK so UTC+1.

**Comment by bestsauce** [comment]
I updated the [instructions for using MathJax with Github](https://github.com/worstsauce/study/blob/master/README.md), Chrome users can simply just install [this add-on](https://chrome.google.com/webstore/detail/github-with-mathjax/ioemnmodlmafdkllaclgeombjnmnbima). Also this add-on doesn't have the problem outlined earlier (about the need to refresh pages).

**Comment by MaxisJaisi** [comment]
@bestsauce Just to clarify, solutions to exercises will be posted on each Exercise issue right? For example, if I want to do problems 1-10 in chap 1, my "written solutions" will be posted as comments.

**Comment by bestsauce** [comment]
@MaxisJaisi It's not necessary that  you write *all* of your solutions there. Just put the ones that you're not sure about whether they're correct and then put the `question` label so that we can check them (and if you want some hint on an exercise for example put the `help needed` label along with which exact exercise you want help with). Is it clear for you now? :)

**Comment by testTEEEEEEST** [comment]
Test

**Comment by bestsauce** [comment]
Test reply

**Comment by bestsauce** [comment]
Another test reply

**Comment by MaxisJaisi** [comment]
Curly braces {} don't seem to work on MathJax. `\{\}` doesn't work, or am I missing something silly?

**Comment by bestsauce** [comment]
@MaxisJaisi Indeed, let me see if there's a solution to that.

**Comment by bestsauce** [comment]
@MaxisJaisi Not the best solution but it works: replace `\` with `&#92;`, that way it seems to work: $&#92;{&#92;}$

**Comment by bestsauce** [comment]
@MaxisJaisi Found out one solution, replace `\` with `&#92;`, for example: $&#92;{ &#92;}$ (don't forget to leave one space before `&#92;` otherwise it doesn't work)

**Comment by bestsauce** [comment]
@MaxisJaisi Found out one solution, replace `\` with `&#92;`, for example: $&#92;{&#92;}$.

Not the best solution but it works, if you happen to find out a better one please let me know.

**Comment by bestsauce** [comment]
@MaxisJaisi Found out a better solution: Use two `\` instead of one, and it should work: `$\\{\\}$` renders as: $\\{\\}$.

**Comment by bestsauce** [comment]
Test (ignore this) $\{\}$

**Comment by MaxisJaisi** [comment]
@bestsauce Alternative one could just do away with MathJax when it comes to braces. 

$\ \mathbf{R}$ $\cup$  {$\infty$} $ \cup$ {$-\infty$}

Doesn't look too bad, does it? (= Anyway `$\\{\\}$`is the best workaround, I believe. Thanks!

**Comment by bestsauce** [comment]
@MaxisJaisi Not too bad but then you have to write many $ to get it right, writing \\ is faster.

**Comment by MaxisJaisi** [comment]
@bestsauce New chapter, today or tomorrow?

**Comment by bestsauce** [comment]
@MaxisJaisi Tomorrow :)
Will finish what's left on my notes and the rest of the exercises today. In the meantime we may have someone who would want to join the group.

**Comment by MaxisJaisi** [comment]
By the way, I transferred part of my _LaTeX_ed notes I wrote down while reading the book onto the Notes section. They are quite brief (I didn't include the stock of examples I collected), require stylistic improvements, and they might contain mistakes. Certain proofs may be clumsy, so suggestions for improvement are appreciated.

**Comment by keyehzy** [comment]
I want to hope in this study group. I am still about to start chapter 1 but don't want to slow you down.

**Comment by bestsauce** [comment]
@masamu I sent you an invitation, go to https://github.com/worstsauce to accept it :smile: 
Once done, follow the steps here https://github.com/worstsauce/study/issues/1#issuecomment-294166204

By the way, did you get MathJax to work in Github?

**Comment by bestsauce** [comment]
@MaxisJaisi Thanks for that! 
Do you think it would be better to have a specific folder in the repository with your notes and exercises (so as to keep the issues pages distinct)? (Here's an example: https://github.com/worstsauce/study/blob/master/resources/Chapter%201/Notes-Maxis.md )

**Comment by keyehzy** [comment]
@bestsauce Yes, I'm set up and ready to go! Thanks.

**Comment by bestsauce** [comment]
@masamu Great! I already made two issue threads for you so that you can track your progress (you can change the one on exercises by deleting the exercises that you do not plan to make if you want)

https://github.com/worstsauce/study/issues/10
https://github.com/worstsauce/study/issues/9

If you need any help with regards to exercises or you have a question regarding a theorem or an exercise you may add the "question" or "help needed" labels in the respective thread with a comment detailing your issue and we'd be able to help.

Also is the May 7 milestone ok for you?

**Comment by MaxisJaisi** [comment]
@bestsauce Yes, a separate folder would be good; my issues look quite cluttered with the notes and exercises thrown in.

**Comment by bestsauce** [comment]
@MaxisJaisi Ok, I'll move them there.

**Comment by MaxisJaisi** [comment]
Where should I put my notes for Chapter 2?

**Comment by bestsauce** [comment]
@MaxisJaisi I've created https://github.com/worstsauce/study/blob/master/resources/Chapter%202/Notes-Maxis.md for you, just edit it and save

**Comment by MaxisJaisi** [comment]
What should we do about @masamu ? He seems to have disappeared...

**Comment by bestsauce** [comment]
@MaxisJaisi  A bit unfortunate that he didn't proceed, I'll leave his two issues open for a while...

**Comment by MaxisJaisi** [comment]
@bestsauce  When typing out a solution to a problem, the equation

$T(x_1, \dots, x_m) = T(x_1(1,0,\dots,0)+x_2(0,1,\dots,0)+\dots+x_m(0,0,\dots,1)) = T(x_1(1,0,\dots,0))+T(x_2(1,0,\dots,0))+\dots+T(x_m(0,0,\dots,1)) = 
x_1 T(1,0,\dots,0)+x_2 T(0,1,\dots,0)+\dots+x_m T(0,0,\dots,1)$

overflows to the right. Any idea on how to fix this?

**Comment by MaxisJaisi** [comment]
Also, how do you write piecewise functions here?

**Comment by bestsauce** [comment]
$$T(x_1, \dots, x_m) = T(x_1(1,0,\dots,0)+x_2(0,1,\dots,0)+\dots+x_m(0,0,\dots,1)) = T(x_1(1,0,\dots,0))+T(x_2(1,0,\dots,0))+\dots+T(x_m(0,0,\dots,1)) = 
x_1 T(1,0,\dots,0)+x_2 T(0,1,\dots,0)+\dots+x_m T(0,0,\dots,1)$$

**Comment by bestsauce** [comment]
@MaxisJaisi Only solution it seems would be to use `\begin{align} ... \end{align}` 

```
$\begin{align}
T(x_1, \dots, x_m) & = T(x_1(1,0,\dots,0)+x_2(0,1,\dots,0)+\dots+x_m(0,0,\dots,1)) \\\\
& = T(x_1(1,0,\dots,0))+T(x_2(1,0,\dots,0))+\dots+T(x_m(0,0,\dots,1)) \\\\
& = x_1 T(1,0,\dots,0)+x_2 T(0,1,\dots,0)+\dots+x_m T(0,0,\dots,1) \\\\
\end{align}$
```

Result

$\begin{align}
T(x_1, \dots, x_m) & = T(x_1(1,0,\dots,0)+x_2(0,1,\dots,0)+\dots+x_m(0,0,\dots,1)) \\\\
& = T(x_1(1,0,\dots,0))+T(x_2(1,0,\dots,0))+\dots+T(x_m(0,0,\dots,1)) \\\\
& = x_1 T(1,0,\dots,0)+x_2 T(0,1,\dots,0)+\dots+x_m T(0,0,\dots,1) \\\\
\end{align}$

----------------------

> Also, how do you write piecewise functions here? 

```
\begin{cases}
x^2, & x\in\mathbf Q \\\\
x, & x\notin\mathbb Q
\end{cases}
```

Result

$$\begin{cases}
x^2, & x\in\mathbf Q \\\\
x, & x\notin\mathbb Q 
\end{cases}$$

**Comment by MaxisJaisi** [comment]
@bestsauce Thanks. But what if I want to write $f(x)$ exactly to the left of the piecewise requirement?

**Comment by MaxisJaisi** [comment]
@bestsauce Nevermind, got it.

**Comment by bestsauce** [comment]
@MaxisJaisi Tried to transform your notes into a nice website but after a lot of trying it seemed to be not worth the effort,
![](https://i.stack.imgur.com/IUsCm.png)

I take my notes by hand, but if you want you can just upload the .tex files and I'll edit them which should be more efficient than transforming them into markdown.

**Comment by MaxisJaisi** [comment]
@bestsauce Sorry for the silence. I was caught up with a lot of things, and took a break (was on a holiday with my family throughout the past week). I'll send you the TeX files once I'm done with Chapter 3. 

By the way, we seem to have slowed down quite a bit, and it's not clear if the whole project is still going forward. I've learned a lot from this, and feel that it's a shame if this group dies out. I suppose we could discuss on reorienting ourselves a bit? Chapter 3 is a big chapter, and it's very important. I think completing Chap 3 would make studying the subsequent chapters much smoother, so it's crucial that we finish it.

**Comment by bestsauce** [comment]
@MaxisJaisi 

> By the way, we seem to have slowed down quite a bit, and it's not clear if the whole project is still going forward. I've learned a lot from this, and feel that it's a shame if this group dies out. I suppose we could discuss on reorienting ourselves a bit? Chapter 3 is a big chapter, and it's very important. I think completing Chap 3 would make studying the subsequent chapters much smoother, so it's crucial that we finish it.

Yeah sorry for that, I was pretty busy these weeks and didn't have much free time. How about we take a one week pause (instead of constantly pushing the milestone) and then be more strict with the schedule (I'll have much more free time after that so I can dedicate the necessary time)?

And yes, I think after finishing Chapter 3 we can move on a lot faster.

**Comment by MaxisJaisi** [comment]
@bestsauce That sounds good to me.

**Comment by bestsauce** [comment]
@MaxisJaisi Ok, I updated the milestone now!

**Comment by celiopassos** [comment]
I'm interesting in participating. I have been using [this site](http://linearalgebras.com) to check my answers, but some are either missing or proven differently. I look forward to post some of my solutions here and see if you guys can point out a mistake I could've possibly made.

**Comment by bestsauce** [comment]
You're welcome @cpassos ! :)
I sent you an invitation, visit https://github.com/worstsauce to accept the invitation.
Once done, do you want to start working on chapter 3 (which we work on right now)?

**Comment by celiopassos** [comment]
@bestsauce Thanks. Chapter 3 is just the one I'm on right now. How does this work? Everyone has a separate issue with exercises?

**Comment by bestsauce** [comment]
@cpassos 

There are two `Chapter x` issues, one with the `Notes` label and the other with `Exercises` label. As you study along you may add the additional `help wanted` label to get some help (e.g. hint), or `question` if it's just a question (e.g. can anyone check if my solution is correct? I don't understand this part of the proof, can anyone clarify it?), and remove them once you're done. 

Here are you two issues for chap3: https://github.com/worstsauce/study/issues/20 and https://github.com/worstsauce/study/issues/19

In addition, we still have a question that needs to be answered for chap1 :) https://github.com/worstsauce/study/issues/8

Also here's a short tutorial on how to get MathJax working with Github: https://github.com/worstsauce/study#how-to-mathjax--github

**Comment by MaxisJaisi** [comment]
@cpassos If you're into writing notes for yourself, you could perhaps share them here. I did it for Chap 1 & Chap 2, but they are incomplete, and devoid of good examples. As you can see from our discussion I'm planning to send my TeXed notes to @bestsauce; he'll work his magic on it, and we'll see what happens. 

What's your (academic) background, if you don't mind? Why linear algebra?

Also, I'm guessing, from your name, that you're Italian... ?

**Comment by celiopassos** [comment]
@MaxisJaisi I'm from Brazil. I just started my major in math. I took interest in Axler's book due to the way it is laid out (everything is detailed and no determinants until the end) and so I could develop my proof writing skills.

**Comment by bestsauce** [comment]
Well looks like we're stalling again...

**Comment by celiopassos** [comment]
@bestsauce You guys are, I'm not, I will finish Chapter 3 in 2 weeks. I'm doing section F now, after that I'll redo section E, because I didn't actually go through all of it, and after that I'll just review C and D.

**Comment by MaxisJaisi** [comment]
@bestsauce 

It's interesting to think about why exactly we haven't been keeping up with the schedule. I've seen many online study groups that have failed, and always wondered why. Now that I've been in one, I can only speculate, and these reasons will be personal. I think it may help to reflect on why there was lack of progress:

1. **Number of people engaged in the program**. There were only two of us, and our interaction had been quite impersonal, so it was difficult to maintain the momentum. There are not enough people to "push" the program forwards, so to speak: it's all too easy to stall when there are only two people, because we could always push the deadline forwards. During the early stages of our group study, I would procrastinate, and then towards the end of the week I'd see @bestsauce doing more problems and asking more questions, and I'd naturally attempt the problems on my own and try to catch up. Without this impetus I found myself working on other things. 

2. **Being engaged in another demanding activity**. I am working through a book on real analysis, and when I get to a difficult portion of the text, I find myself not being able to disengage from thinking about it, and this leads to my neglecting linear algebra. I guess this is personal, because some are better at multitasking than others, but I stubbornly refuse to budge when there's an unresolved problem. 

3. **It's difficult to start after a period of inactivity**. This goes without saying, but I'm finding it very difficult to set my mind straight to think about linear maps between vector spaces after several weeks of abandoning the subject. This is in direct contrast with the momentum I had when I focused all my time on chapter 1 and chapter 2. I think extending deadlines isn't a healthy thing to do... We should at least move forwards, even though we haven't fully mastered previous concepts. We can always go back and fill in the details in the future. 

4. **"Perfecting notes"**. I spent way too much time thinking about how to rewrite linear algebra concepts in my own terms, to the extent that I neglected the problem sets. I think I should focus more on the problems from now on, and write the notes only after finishing a chapter. Also, I spent an inordinate amount of time worrying about "getting the perfect picture of the subject" by consulting way too many sources and digging too deeply, which isn't exactly the wisest thing to do, considering the fact that this is my first exposure to the subject.

5. **Lack of previous exposure to the subject**. This bothered me the most. Although Axler's approach is very aesthetic and clean, I found myself limited by my lack of exposure to elementary computational linear algebraic concepts, like manipulating matrices, solving linear equations, row operations, and the connection between properties of matrices and the maps they represent. I then spent a lot of time practising these things using other sources, to the detriment of my progress in this program. 

Anyway, I hope @cpassos will stay, because three is better than two. I'll try my best to catch up in these 2 weeks...

**Comment by MaxisJaisi** [comment]
@cpassos Again, out of curiosity, are you taking a linear algebra course at uni currently?

**Comment by celiopassos** [comment]
MaxisJaisi I will definitely stay if you do as well.

In my school, we have just finished a class they call "Analytic Geometry and Linear Algebra". I think it is supposed to just give a geometric intuition about linear algebra concepts, since there was no rigour (we didn't see a single proof in the entire program). Unfortunately, our professor was pretty bad, so we didn't develop intuition either, just made some mindless computations.

**Comment by MaxisJaisi** [comment]
@cpassos I will. Lack of a proper background in LA cripples me severely when it comes to learning interest and important mathematics (chiefly analysis of several real variables, calculus on manifolds...), so I definitely must finish Axler this year. I just lost momentum, that's all. Will get back to Chap 3 once I tidy up a few things I'm doing in analysis today.

**Comment by bestsauce** [comment]
@MaxisJaisi I think you can add that we set all exercises to make, which is too demanding (with respect to our schedule). I think we should be more selective.

Anyway, glad we're still alive, going to reset the milestone.

**Comment by MaxisJaisi** [comment]
@bestsauce Do you agree that we should open an issue for 3.C & 3.D? I'm about halfway through the exercises in 3.B, but also completed sections 3.C & 3.D (though I haven't touched the exercises). Still trying to internalise quotient and dual spaces, but I think I can multitask. Don't close the 3.B issue first.

**Comment by celiopassos** [comment]
Guys, I'll be posting my solutions only on this [repo](https://github.com/cpassos/linear-algebra-done-right-solutions) and not on the issues, because I like to keep each statement on its own line but GitHub renders newlines differently in issues (which make my solutions look weird) and I don't I want to have to keep two versions up to date.

Nevertheless, I will still follow what you guys post here and create issues here to post questions, just not the solutions.

**Comment by celiopassos** [comment]
Looks like @bestsauce has deserted us.

**Comment by MaxisJaisi** [comment]
@cpassos Sounds good. Actually the solutions posted on issues are temporary. My solutions were transferred to separate repositories after completing the previous 2 chapters. I have thoughts about going back and cleaning up my solutions after I'm done with the whole book, since in hindsight, I realise that some of my previous solutions were quite drawn out and confusing.

By the way, nearly done with the exercises in 3B, and will get to the problem you asked soon. After finishing 3B I'll jump ahead to 3E & 3F, because I'm least comfortable with these chapters ( still trying to picture quotient spaces... ). And then after I'm done with chapter 3, I'll skip the polynomials chapter and head straight to eigenvectors & eigenvalues. What do you think? If you feel you want to start with chapter 4 and chapter 5, you could do so, I'll trail behind, but it's still possible to cross check our solutions. 

Perhaps @bestsauce is busy?

**Comment by celiopassos** [comment]
@MaxisJaisi I'm also finding quotients spaces rather difficult, I still couldn't do the exercises. Axler says in the preface that this chapter, and the duality one, is a bit abstract and can be skipped without running into problems later in the book. Perhaps we should put them aside for now and come back at them later once we develop more intuition. This strategy works well for me when I find something too difficult. I always discover new insight when I look at something a second time.

About chapter 4, I think going through it is worth it, since it is so short. Also, some of the results there are proved using linear algebra, which is nice.

**Comment by MaxisJaisi** [comment]
@cpassos Okay, I'll do that --- skip quotient spaces and duality. Will finish up the remaining exercises in the chapter, and then go to polynomials.

**Comment by MaxisJaisi** [comment]
@cpassos I'll be away for a week (preparing for an entrance exam). Will be back when it ends!

**Comment by celiopassos** [comment]
@MaxisJaisi Ok.

**Comment by MaxisJaisi** [comment]
Done with exams.

@cpassos, I just read your answer to Exercise 24, 3.B. It seems that it can be simplified considerably. First, do the same thing as you did (take a preimage set) for $T_{1}$, construct the map in the same way, call it $S$. Then the linear map we want is $T_{2} \circ S$. We verify that $T_{2} \circ S \circ T_{1} = T_{2}$ by noting that $S(T_{1} (v)) - v \in \operatorname{null} (T_{1})$, for all $v \in V$. 

Also, you assumed that $\operatorname{null} T_{1} = \operatorname{null} T_{2}$, but the question only says that one is a subset of the other, not that they are equal.

**Comment by celiopassos** [comment]
@MaxisJaisi 

> Then the linear map we want is $T_2 \circ S$.

We want just $T_2$. $T_2 S$ is in exercise 25.

> We verify that $T_2 \circ S \circ T_1 = T_2$ by noting that $S(T_1(v)) − v \in \operatorname{null}(T_1)$, for all $v \in V$.

I guess $T_2 \circ S \circ T_1$ doesn't make sense because the domain of $T_2$ is $V$, but the codomain of $S$ is $W$. Perhaps you mean to construct $S: \operatorname{range} T_1 \to V$ such that $Sw_j = v_j$? Could you elaborate this a bit more? I can't see how to prove that part you note without first showing that $V = \operatorname{null} T_1 + \operatorname{span}(v_1, \dots, v_n)$.

> Also, you assumed that $\operatorname{null} T_1 = \operatorname{null} T_2$, but the question only says that one is a subset of the other, not that they are equal.

Thanks, I fixed it. This mistake did not influence the rest of the proof though, because no part in it required that $\operatorname{null} T_2 \subset \operatorname{null} T_1$.

**Comment by MaxisJaisi** [comment]
@cpassos I apologise for the lousy write-up, I typed things up on the fly, and was rather jaded after the exams. Anyway, let me explain it again, hopefully more carefully this time:

I'll paraphrase Axler's question again.

> Let $T_{1}, T_{2}$ be linear maps from $V$ to $W$, where $W$ is finite-dimensional. Then $\operatorname{null} T_{1} \subset \operatorname{null} T_{2}$ only if there exists a linear operator $S$ on $W$ such that $ST_{1} = T_{2}$.

The main intuition (I think) is that if $T_{1}$ is invertible, then it's easy to construct $S$: We first send everything back, from $W$ to $V$, by applying the inverse of $T_{1}$, let's call it $F$,  to $T_{1}$. We get $F \circ T_{1} = e_{V}$. Then we easily have, by applying $T_{2}$ on the left, $T_{2} \circ F \circ T_{1} = T_{2}$.

But $T_{1}$ isn't invertible, so we have a problem. We get around this problem by constructing a "partial inverse" of $T_{1}$: we first specify a basis $w_{1}, w_{2}, \dots, w_{n}$ for $\operatorname{range}(T_{1})$. Then there must be a $v_{i} \in V$ such that $T_{1} (v_{i}) = w_{i}$, for $1 \leq i \leq n$ (note that this doesn't involve the axiom of choice. in general, it does, if we don't restrict $W$ to be finite-dimensional, but Axler wants to avoid choice). Note also that we're fixing $v_{i}$, by specifying a single preimage from the set of preimages for each basis $w_{i}$. We build a "left partial inverse" $L$, acting on $W$, as follows: $L(w_{i}) = v_{i}$ for each $1 \leq i \leq n$, and extend linearly to the entire basis of $W$. 

Then I claim that $T_{2} \circ L$ will be the $S$ that we want. This is where the hypothesis $\operatorname{null}(T_{1}) \subset \operatorname{null}(T_{2})$ comes in. To see that $T_{2} \circ L \circ T_{1} = T_{2}$, consider $v \in V$, and we compute:

$S \circ T_{1} (v) = T_{2} \circ L \circ T_{1} (v) = T_{2} (L(T_{1} (v))) = T_{2} (L(a_{1} w_{1} + a_{2} w_{2} + \dots + a_{n} w_{n})) = T_{2} (a_{1} v_{1} + a_{2} v_{2} + \dots + a_{n} v_{n})$. 

Now if we can show that $T_{2} (a_{1} v_{1} + a_{2} v_{2} + \dots + a_{n} v_{n}) = T_{2} (v)$, we are done. The expression is equivalent to $T_{2} (a_{1} v_{1} + a_{2} v_{2} + \dots + a_{n} v_{n}) - T_{2} (v) = \mathbf{0}$, which is equivalent to $ T_{2} (v) - T_{2} (a_{1} v_{1} + a_{2} v_{2} + \dots + a_{n} v_{n}) = \mathbf{0}$, which is equivalent to (since $T_{2}$ is linear), $ T_{2} (v - (a_{1} v_{1} + a_{2} v_{2} + \dots + a_{n} v_{n})) = \mathbf{0}$, which is equivalent to $v - (a_{1} v_{1} + a_{2} v_{2} + \dots + a_{n} v_{n}) \in \operatorname{null} (T_{2})$, which is true, because $\operatorname{null}(T_{1}) \subset \operatorname{null}(T_{2})$! (I'll leave it to you to verify this last step)

Well, the writeup was rather painful, so let's rewrite it in a more streamlined fashion, which might confuse someone encountering it for the first time:

**Proof**: We first build a partial inverse, $L$ of $T_{1}$ as follows: let $w_{1}, \dots, w_{n}$ be a basis for $\operatorname{range}(T_{1})$ (this enumeration is possible, since $W$ is finite-dimensional.) Given each $w_{i}$ ($1 \leq i \leq n$), we pick a corresponding $v_{i} \in V$ such that $T_{1} (v_{i}) = w_{i}$. Then define $L$ on $W$ by $L(w_{i}) = v_{i}$ ($1 \leq i \leq n$), and extend linearly to a full basis of $W$.

I claim that $S = T_{2} \circ L$ works, that is, $T_{2} \circ L \circ T_{1} = T_{2}$. To see this, we first note that for any $v \in V$, we have $L(T_{1} (v)) - v \in \operatorname{null} (T_{1}) \subset \operatorname{null} (T_{2})$, so $T_{2} ((L(T_{1} (v)) - v) = \mathbf{0}$, which is equivalent to $T_{2} \circ L \circ T_{1} (v) = T_{2} (v)$.

**Comment by MaxisJaisi** [comment]
@cpassos Strange, the paragraph after the quoted question doesn't render correctly, and I can't seem to work out what went wrong. Anyway, you can ignore it for now, it doesn't affect the rest of the argument, and I'll try fixing it afterwards.

**Edit**: Fixed

**Comment by celiopassos** [comment]
Thanks! That is really nice, much shorter than mine and avoids altogether the $V = \operatorname{null} T_1 + \operatorname{span}(v_1, \dots, v_n)$ thing.
